alb.tf

```
@@ -0,0 +1,131 @@
+resource "aws_security_group" "alb_https" {
+  count = var.create_alb ? 1 : 0
+
+  name_prefix = var.name
+  description = "Bitbucket Inbound LB (HTTPS)"
+  vpc_id      = var.vpc_id
+
+  tags = merge(
+    var.tags,
+    var.alb_additional_sg_tags,
+    { "Name" : var.name }
+  )
+
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+resource "aws_security_group_rule" "alb_https_egress" {
+  count = var.create_alb ? 1 : 0
+
+  description              = "Allow traffic from the LB to the instances"
+  from_port                = 7990
+  protocol                 = "tcp"
+  security_group_id        = aws_security_group.alb_https[0].id
+  source_security_group_id = aws_security_group.this.id
+  to_port                  = 7990
+  type                     = "egress"
+}
+
+
+resource "aws_security_group_rule" "alb_https_ingress" {
+  count = var.create_alb && length(var.alb_allowed_https_cidr_blocks) > 0 ? 1 : 0
+
+  cidr_blocks       = var.alb_allowed_https_cidr_blocks #tfsec:ignore:AWS006
+  description       = "Allow HTTPS traffic from the allowed ranges"
+  from_port         = var.alb_https_port
+  protocol          = "tcp"
+  security_group_id = aws_security_group.alb_https[0].id
+  to_port           = var.alb_https_port
+  type              = "ingress"
+}
+
+resource "aws_lb" "https" {
+  count = var.create_alb ? 1 : 0
+
+  name_prefix     = substr(var.name, 0, 6)
+  internal        = var.alb_https_internal
+  security_groups = [aws_security_group.alb_https[0].id]
+  subnets         = var.alb_https_subnets
+  tags            = var.tags
+
+  dynamic "access_logs" {
+    for_each = var.access_logs_enabled ? ["this"] : []
+
+    content {
+      bucket  = var.access_logs_bucket
+      prefix  = var.access_logs_prefix
+      enabled = var.access_logs_enabled
```

You're already only creating the block if this var is true, so there's no need to evaluate the truthiness again

```suggestion
      enabled = true
```